### PR TITLE
Unit test scheduler framework extenders fix

### PR DESCRIPTION
**Pull request by RyanBouchard**
#### What type of PR is this?
/kind cleanup
/sig scheduling

#### What this PR does / why we need it:
This PR adds a test to cover the bug fix that addresses scheduler extenders not being called on preemptions because frameworkImpl.extenders is not set.
#### Which issue(s) this PR fixes:

Follow-up of https://github.com/kubernetes/kubernetes/pull/103019#issuecomment-864678353

#### Special notes for your reviewer:
NONE

#### Does this PR introduce a user-facing change?
NONE

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by RyanBouchard** [comment]
@Huang-Wei tagging you as requested.

**Comment by alculquicondor** [comment]
The request was for integration tests.
I think the unit test added in #103019 was enough.
https://github.com/kubernetes/kubernetes/blob/master/test/integration/scheduler/extender_test.go

File: pkg/scheduler/extender_test.go [hunk 1]
```diff
@@ -252,6 +257,35 @@ func TestGenericSchedulerWithExtenders(t *testing.T) {
 			},
 			name: "test 9",
 		},
+		{
+			registerPlugins: []st.RegisterPluginFunc{
+				st.RegisterFilterPlugin("TrueFilter", st.NewTrueFilterPlugin),
+				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
+				st.RegisterPostFilterPlugin(defaultpreemption.Name, func(plArgs apiruntime.Object, fh framework.Handle) (framework.Plugin, error) {
```
Comment (Huang-Wei) [inline pkg/scheduler/extender_test.go:265]: Hmm... the original idea was to verify the registered extenders are passed all the way to defaultpreemption plugin, and also **get called**.

But this UT only tests on `schedule.Schedule()`, which doesn't proceed for run PostFilter plugins, and hence cannot verify if the extenders are called or not:

https://github.com/kubernetes/kubernetes/blob/9ff3b7e744b34c099c1405d9add192adbef0b6b1/pkg/scheduler/scheduler.go#L540

We may pursue it in some test (e.g., pkg/scheduler/scheduler_test.go) that covers `scheduleOne()`, or directly test `RunPostFilterPlugins()`. (the previous option works better)

**Comment by Huang-Wei** [comment]
> I think the unit test added in #103019 was enough.

@alculquicondor Not quite, the test in that PR ensured the extenders were present, but didn't verify they get called via the framework in defaultpreemption plugin, in a stateful manner. The current scheduler/extender_test.go verified _partial_ paths that extenders get called, in particular, `findNodesThatPassExtenders()`  in a stateless manner by passing in the extenders. (it didn't pass in extenders via `runtime.WithExtenders(extenders)` to the framework instance)

> The request was for integration tests.

If a UT can cover the logic, I'm fine to not pursue it in an integration test. Commented in https://github.com/kubernetes/kubernetes/pull/104092#discussion_r681978903, let's see if a UT can qualify.

**Comment by alculquicondor** [comment]
I see. Sounds good

**Comment by RyanBouchard** [comment]
Hello, thank you for the feedback. I will pursue addressing this in pkg/scheduler/scheduler_test.go with a UT.

**Comment by dims** [comment]
is this still needed, please rebase if we still need it (or we can close it?)

**Comment by dims** [comment]
Is this PR still needed, please rebase if so (or we can close it?)

**Comment by RyanBouchard** [comment]
@dims @Huang-Wei @alculquicondor I am apologize for the long absence. Other things have since come up, and I don't think I will get to this in a reasonable amount of time. I will be closing this PR to avoid holding things up further.

**Comment by Huang-Wei** [comment]
@RyanBouchard never mind. Feel free to open a new PR when it's ready.

**Pull request closed without merging**